ibis/expr/datatypes.py

```diff
@@ -305,6 +305,12 @@ def valid_literal(self, value):
         return isinstance(value, six.string_types)
 
 
+class Binary(Variadic):
```

Comment:
May be worth now adding a docstring to contrast string vs. binary. Though the definition of "string" may vary from back end to back end (e.g. we cannot necessarily rely on it being utf8)